sdk/core/Azure.Core.Experimental/src/PopTokenRequestContext.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Contains the details of an authentication token request.
+    /// </summary>
+    public readonly struct PopTokenRequestContext
+    {
+        /// <summary>
+        /// Creates a new TokenRequest with the specified scopes.
+        /// </summary>
+        /// <param name="scopes">The scopes required for the token.</param>
+        /// <param name="parentRequestId">The <see cref="Request.ClientRequestId"/> of the request requiring a token for authentication, if applicable.</param>
+        /// <param name="claims">Additional claims to be included in the token.</param>
+        /// <param name="tenantId">The tenant ID to be included in the token request.</param>
+        /// <param name="isCaeEnabled">Indicates whether to enable Continuous Access Evaluation (CAE) for the requested token.</param>
+        /// <param name="proofOfPossessionNonce">The nonce value required for PoP token requests.</param>
+        /// <param name="request">The request to be authorized with a PoP token.</param>
+        public PopTokenRequestContext(string[] scopes, string? parentRequestId = default, string? claims = default, string? tenantId = default, bool isCaeEnabled = false, string? proofOfPossessionNonce = default, Request? request = default)
+        {
+            Scopes = scopes;
+            ParentRequestId = parentRequestId;
+            Claims = claims;
+            TenantId = tenantId;
+            IsCaeEnabled = isCaeEnabled;
+            ProofOfPossessionNonce = proofOfPossessionNonce;
+            _request = request;
+        }
+
+        /// <summary>
+        /// Creates a new TokenRequestContext from this instance.
+        /// </summary>
+        /// <returns></returns>
+        public TokenRequestContext ToTokenRequestContext()
```

Comment:
Nit: .NET naming convention tends to be `As` instead of `To`.

Reply:
...or even `GetTokenRequestContext()`.

Reply:
also temporary. Going to leave for now.